tests/test_config.py

```diff
@@ -435,17 +446,37 @@ def test_merge():
         else:
             assert config2[key] == valid_params[key][2]
 
-    # If conflicts, kwargs takes precedence
+    # If conflicts, kwargs takes precedence ...
     kwargs['ra_col'] = 'alpha2000'
     config2 = treecorr.config.merge_config(config1, kwargs, treecorr.Catalog._valid_params)
     assert config2['ra_col'] == 'alpha2000'
 
+    # ... even when when the value in kwargs is None.
+    kwargs = {'bin_size': 0.06, 'nbins': None}
+    config2 = treecorr.config.merge_config(config1, kwargs, treecorr.BinnedCorr2._valid_params)
+    assert config2['bin_size'] == 0.06
+    assert config2['min_sep'] == config1['min_sep']
+    assert config2['max_sep'] == config1['max_sep']
+    assert kwargs['nbins'] is None
+    assert config2['nbins'] is None
+
```

Comment:
Okay, I think it now does what a user would intuitively expect, and the tests seem to pass.